modules/lazyload/module.php

```diff
@@ -92,13 +89,23 @@ private function append_lazyload_selector( $control, $value ) {
 				if ( 0 === strpos( $css_property, 'background-image' ) ) {
 					if ( ! empty( $value['url'] ) ) {
 						$control['selectors'][ $selector ] = $css_property . '--e-bg-lazyload: url("' . $value['url'] . '");';
+						$control = $this->apply_dominant_color_background( $control, $value, $selector );
 					}
 				}
 			}
 		}
 		return $control;
 	}
 
+	private function apply_dominant_color_background( $control, $value, $selector ) {
+		$metadata = wp_get_attachment_metadata( $value['id'] );
+		$dominant_color = Utils::get_array_value_by_keys( $metadata, [ 'dominant_color' ] );
+		if ( $dominant_color ) {
+			$control['selectors'][ $selector ] .= "background-color: #{$dominant_color};";
```

Comment:
What if the user already has a background color?

Reply:
The task said : Use dominant colors if possible, and make sure they won’t be generated on the fly (on page request to avoid an increased TTFB). Therefore, I think we should ask @avivu